src/main/java/io/github/phantamanta44/libnine/client/gui/component/impl/GuiComponentTextInput.java

```diff
@@ -23,12 +24,17 @@
     private String value;
     @Nullable
     private final String tooltipKey;
+    private FieldType fieldType = FieldType.ALL_TEXT;
```

Comment:
field type should not be mutable

Reply:
You're right. It's inconvenient without a builder instead of just constructors though